### ISSUE TITLE
Play Track - override event from the same input source

### ISSUE THREAD

**Issue by milosfec** [FrSkyRC/ETHOS-Feedback-Community#2397]
# Current behaviour
- all track are added into queue
- all added track are going to be played
- when playing track for mode change assigned to 6 position switch, it plays all tracks:
  - Switch goes from 1 to 6: "Flight mode 2", "Flight mode 3", "Flight mode 4", "Flight mode 5", "Flight mode 6"
  - Switch goes from 6 to 1: "Flight mode 5", "Flight mode 4", "Flight mode 3", "Flight mode 2", "Flight mode 1"
  - When new event occurs that is suppose to play a track, it play's it usually at the end (can be somewhere in between)
  - Rotate switch multiple times and listen to all track for a long time...

# Expected behaviour
- add an option to override track form the same input source
- when playing track for mode change assigned to 6 position switch, it should:
  - Switch goes from 1 to 6:
    - Acceptable: "Flight mode 2", "Flight mode 6"
    - Preferred: "F", "F", "F", "F", "Flight mode 6"
    - Extra feature: allow setting delay before playing and it would play just "Flight mode 6"
  - If new event occurs while rotating the switch:
    - Acceptable: "Flight mode 2", "Some new event", "Flight mode 6"
    - Preferred: "F", "F", "Some new event", "F", "F", "Flight mode 6"
    - With extra feature (delay): "Some new event", "Flight mode 6"

It should be possible to achieve this behaviour using logical switches, but it would require setting 6 logical switches, then 6 special functions, and it doesn't allow using global function as there are no global logical switches.

**Comment by mawzthefinn** [comment]
What's your Switch Middle Detect Delay set to? That's in System->Hardware->Switches

Setting to a value higher than 0ms (the default) should resolve that, as it requires the switch to be in a specific position for a moment before it goes true. 0ms is only required for self-check on old S6R and S8R receivers.

**Comment by milosfec** [comment]
Good idea, that works! But for "panic" switch (stabilisation) I don't want any delay. In that case it would be good to have a feature to override specific switch delay. For 3 position switches like stabilisation mode <100ms is fine, for 6 position even 500ms may be good, but 0ms for panic and throttle cut...

**Comment by mawzthefinn** [comment]
Even for panic or throttle cut, you won't even notice 100ms. 500ms is probably too long

**Comment by milosfec** [comment]
I had a close call where panic switch saved my model ~20cm from the ground. 100ms delay would definitely meant crash.

**Comment by iflyforfood** [comment]
> What's your Switch Middle Detect Delay set to? That's in System->Hardware->Switches
> 
> Setting to a value higher than 0ms (the default) should resolve that, as it requires the switch to be in a specific position for a moment before it goes true. 0ms is only required for self-check on old S6R and S8R receivers.

Was about to start a ticket for this same annoyance but this fixed it. I don't get why it's not enabled by default

**Comment by mawzthefinn** [comment]
The reason why switch middle detect delay defaults to 0ms is that the self-check procedure on the S6R and S8R (and Archer SR's on old firmware) require this setting to properly enter self-check mode.

Early ETHOS builds had a default 100ms middle detect delay, but users were having issues setting up the stabilizers.

I expect to go back to having a delay by default at some point in the future, as Archer 2.1.10 moved self-check off ch12 and into the menu, so up to date Archer's and newer stabilizers don't have that requirement.

**Comment by bsongis-frsky** [comment]
I am not sure that the condition to allow override should be "the same input source". Searching for a better solution. Perhaps it could fit into Play Sequence which has been added to 1.5.0?

**Comment by GiacomoGaleotti** [comment]
> I am not sure that the condition to allow override should be "the same input source". Searching for a better solution. Perhaps it could fit into Play Sequence which has been added to 1.5.0?

i have the same problem, tried with edge logical switches but are out of my abilty to understand.
a solution can be to add a delay to selected switches istead to have it set for all with the same value. 
so you can decide the switch you want to "slow down" leaving everithing else as default 0ms delay
